### PR TITLE
Closes #81 #92 #93 #94 #95 #96 #97 #99: Various updates

### LINKED ISSUE #81: Add snaphot testing exmaple
Way cool Ross!  Just some thoughts on a few things.  I’m not super passionate about any of them.

Other ideas that maybe others can work on:

- snapshot tests example with two examples one works and one errors (this really makes writing tests so much less painful)
- Remotes in DESCRIPTION 

these overkill? I can make into issues if not

_Originally posted by @bms63 in https://github.com/pharmaverse/admiraltemplate/pull/79#pullrequestreview-2091859716_

### LINKED ISSUE #92: Add SVG logos for founding companies
### Please select a category the issue is focused on?

_No response_

### Let us know where something needs a refresh or put your idea here!

See how it was done here
https://github.com/pharmaverse/admiral/issues/2785

### LINKED ISSUE #93: Update LICENSE.md to display as done in extension packages
### Please select a category the issue is focused on?

_No response_

### Let us know where something needs a refresh or put your idea here!

Currently displayed as: 
Copyright `<<INSERT YEAR>>` `<<INSERT CO-DEVELOPMENT PARTNER COMPANIES AS LICENSE HOLDERS>>`

Should be updated to:
Copyright `Year: <<INSERT YEAR>>`  `License holders: <<INSERT CO-DEVELOPMENT PARTNER COMPANIES AS LICENSE HOLDERS>>`

### LINKED ISSUE #94: Documentation:  Use `@permitted` and `@default` custom admiraldev roclets for the example function
### Please select a category the issue is focused on?

Function Documentation

### Let us know where something needs a refresh or put your idea here!

See [here](https://pharmaverse.github.io/admiraldev/reference/rdx_roclet.html). 

No need to showcase the custom `@examplesx` examples sections (see [here](https://pharmaverse.github.io/admiraldev/dev/articles/writing_custom_examples.html)) - here the decision is left to the package maintainer as to whether they want to do it for any of the functions.

### LINKED ISSUE #95: Remove devcontainer
No longer used - remove.

Image

### LINKED ISSUE #96: Update lintr configurations to use central `{admiraldev}` file
Pull request https://github.com/pharmaverse/admiraldev/pull/507 will, once merged, introduce a way to set up custom `{lintr}` configurations in `{admiraldev}` which can then used by all other packages in the `{admiral}` ecosystem. If the configurations are updated in `{admiraldev}`, they will trickle down to the extension packages, unless they are overridden (see below). Currently the configurations we have chosen are:

- line length: 100 (same as previously)
- object usage: NULL (same as previously)
- cyclomatic complexity: 20 (same as previously)
- indentation: NULL (same as previously)
- undesirable functions: `stop()`, `warning()`, `message()`, `ifelse()`, `grepl()` 🆕

To set this up for future extensions:

- Rename the `.lint` config file to `.lintr.R`
- Replace the contents of the file with:
```
library(lintr)

source(system.file("lintr/linters.R", package = "admiraldev"))

linters <- admiral_linters()

exclusions <- list(
  "R/data.R" = Inf,
  "inst" = list(undesirable_function_linter = Inf),
  "vignettes" = list(undesirable_function_linter = Inf)
)
```
- Run `lintr::lint_package()` to check the linters run as expected

### LINKED ISSUE #97: Add link to pharmaverse admiral ecosystem page
[Page](https://pharmaverse.org/e2eclinical/adam/)

Either add in sidebar or at the top of the README.

Image

### LINKED ISSUE #99: Website updates
- [x] Ask AI button
- [x] Dark mode
- [x] Tooltips
- [x] Update navbar height
- [x] Add "Explore templates" page
- [x] Add a logo

### PR DESCRIPTION
**Pull request by manciniedoardo**
Closes #81 
Closes #92 
Closes #93 
Closes #94 
Closes #95 
Closes #96 
Closes #97 
Closes #99

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by manciniedoardo** [comment]
@pharmaverse/admiral finally took some time to update admiraltemplate in line with the last year's updates - please take a look!

File: _pkgdown.yml
Comment (bundfussr) [inline _pkgdown.yml]: Should the labels be updated?

**Review by bundfussr — COMMENTED** [review]
Should we update `.github/workflows/common.yml`?
In `.github/CODEOWNERS` Dinakar is referenced. Should we change this?

File: _pkgdown.yml
Comment (manciniedoardo) [reply _pkgdown.yml]: updated how sorry?

**Comment by manciniedoardo** [comment]
> Should we update `.github/workflows/common.yml`? In `.github/CODEOWNERS` Dinakar is referenced. Should we change this?

yes, good idea to remove mention of @cicdguy.

> Should we update .github/workflows/common.yml?

Do you mean with your new workflows?

File: _pkgdown.yml
Comment (bundfussr) [reply _pkgdown.yml]: In the other packages we use "Join the pharmaverse Slack" and "Report an Issue".

**Comment by bundfussr** [comment]
> > Should we update .github/workflows/common.yml?
> 
> Do you mean with your new workflows?

Yes

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -2,4 +2,3 @@
 # Code owners are automatically requested for review when someone opens a pull request
 # that modifies code that they own. In this instance, cicdguy will be notified of any
 # changes made the files inside the .github/workflows directory.
-.github/workflows/* @cicdguy
```
Comment (bundfussr) [inline .github/CODEOWNERS:5]: Hm, now the comment describes the deleted line.

I think it would be helpful to have an example line. We could use one of our github user ids or something like `@<github user id>`.

Comment (manciniedoardo) [reply .github/CODEOWNERS:5]: sorry - did that too quickly

File: _pkgdown.yml
Comment (manciniedoardo) [reply _pkgdown.yml]: updated thanks

File: .github/CODEOWNERS (hunk 1 above)
Comment (bms63) [reply .github/CODEOWNERS:5]: should we just get rid of the codeowners stuff?

File: data-raw/xx.R
Comment (bms63) [inline data-raw/xx.R:1]: is there any notes on encouraging extensions to move sdtm data to pharmaversesdtm?

File: man/figures/logo.png
Comment (bms63) [inline man/figures/logo.png:1]: whoa this is cool!!!

File: R/my_first_fcn.R [hunk 2]
```diff
@@ -2,14 +2,18 @@
...
 #'
 #' @details In the roxygen documentation you will find tags for family and keywords.
 #' This is to create organized sections for the Reference tab on the pkgdown website.
 #' You can modify the `_pkgdown.yml` as necessary to create appropriate sections as necessary.
 #' Under `./man/roxygen/meta.R`, you will find where to store these family/keywords.
+#' You will also find custom permitted text under `./man/roxygen/rdx_meta.R` to
+#' standardize argument text.
 #'
-#' @return Happy Message
+#' @return A message
```
Comment (bms63) [inline R/my_first_fcn.R:16]:
```suggestion
#' @return The happiest of messages
```

Comment (manciniedoardo) [reply R/my_first_fcn.R:16]: it's not always happy if you select FALSE though!!

File: data-raw/xx.R
Comment (manciniedoardo) [reply data-raw/xx.R:1]: added!

File: .github/CODEOWNERS (hunk 1 above)
Comment (manciniedoardo) [reply .github/CODEOWNERS:5]: maybe ok to keep in case another ext team wants to have?

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # This a CODEOWNERS file, where you can establish code owners.
 # Code owners are automatically requested for review when someone opens a pull request
-# that modifies code that they own. In this instance, cicdguy will be notified of any
+# that modifies code that they own. In this instance, @<github user id> will be notified of any
 # changes made the files inside the .github/workflows directory.
```
Comment (bundfussr) [inline .github/CODEOWNERS]:
```suggestion
# changes made to the files inside the .github/workflows directory.
```

File: .Rbuildignore [hunk 4]
```diff
@@ -14,6 +14,8 @@
 ^admiraltemplate\.Rcheck$
 ^admiraltemplate.*\.tar\.gz$
 ^admiraltemplate.*\.tgz$
-^staged_dependencies.yaml$
+^staged_dependencies.yml$
```
Comment (bundfussr) [inline .Rbuildignore]: I think this line shouldn't be changed but `staged_dependencies.yml` in admiraltemplate should be renamed to `staged_dependencies.yaml`. Then it is in line with the admiral packages and the admiralci workflows.

Comment (manciniedoardo) [reply .Rbuildignore]: makes sense, updated

File: man/figures/open_source_logo.png
Comment (bundfussr) [inline man/figures/open_source_logo.png:1]: Where is this image be used?

Comment (manciniedoardo) [reply man/figures/open_source_logo.png:1]: image

it's just to have a "company" logo

Comment (bundfussr) [reply man/figures/open_source_logo.png:1]: I see. Thanks.

**Review by bundfussr — APPROVED** [review]

**Comment by manciniedoardo** [comment]
apologies @bms63 I somehow imagined you'd approved yesterday and thus i merged the PR, but now i saw you had not. if you have any other comments i'm happy to open another PR.
